pycqed/analysis_v2/spectroscopy_analysis.py

```diff
@@ -31,1507 +23,285 @@ def __init__(self, t_start: str = None,
-        elif fitting_model == 'simple_hanger':
-            fit_fn = fit_mods.HangerFuncAmplitude
-            raise NotImplementedError(
-                'This functions guess function is not coded up yet')
-            # TODO HangerFuncAmplitude Guess
-        elif fitting_model == 'lorentzian':
-            raise NotImplementedError(
-                'This functions guess function is not coded up yet')
-            fit_fn = fit_mods.Lorentzian
-            # TODO LorentzianGuess
-        elif fitting_model == 'complex':
-            raise NotImplementedError(
-                'This functions guess function is not coded up yet')
-            # hanger_fit = VNA_analysis(self.timestamps,
-            #                              do_fitting= True,
-            #                              options_dict= {'fit_options':
-            #                                             {'model':'hanger'}},
-            #                              extract_only= True)
-            # hanger_fit_res = hanger_fit.fit_dicts['reso_fit']['fit_res']
-            # complex_guess = hanger_fit_res.best_values
-
-            # delta_phase = np.unwrap(self.proc_data_dict['plot_phase'])[-1] - /
-            #               np.unwrap(self.proc_data_dict['plot_phase'])[0]
-            # delta_freq = self.proc_data_dict['plot_frequency'][-1] - /
-            #              self.proc_data_dict['plot_frequency'][0]
-            # phase_v = delta_phase/delta_freq
-            # fit_fn = fit_mods.SlopedHangerFuncComplex2
-
-            # TODO HangerFuncComplexGuess
-
-        if len(self.raw_data_dict['timestamps']) == 1:
-            if fitting_model == 'complex':
-                self.fit_dicts['reso_fit'] = {'fit_fn': fit_fn,
-                                              'fit_guess_fn': fit_guess_fn,
-                                              'fit_yvals': {'data': self.proc_data_dict['plot_amp']},
-                                              'fit_xvals': {'f': self.proc_data_dict['plot_frequency']}
-                                              }
-            else:
-                self.fit_dicts['reso_fit'] = {'fit_fn': fit_fn,
-                                              'fit_guess_fn': fit_guess_fn,
-                                              'fit_yvals': {'data': self.proc_data_dict['plot_amp']},
-                                              'fit_xvals': {'f': self.proc_data_dict['plot_frequency']}
-                                              }
-        else:
-            self.fit_dicts['reso_fit'] = {'fit_fn': fit_fn,
-                                          'fit_guess_fn': fit_guess_fn,
-                                          'fit_yvals': [{'data': np.squeeze(tt)} for tt in self.plot_amp],
-                                          'fit_xvals': np.squeeze([{'f': tt[0]} for tt in self.plot_frequency])}
-    def analyze_fit_results(self):
-        pass
-
-
-class ResonatorSpectroscopy(Spectroscopy):
```

Comment:
You cannot remove this, it is being used in QuDev_transmon.find_readout_frequency(). This is the routine we use to find the optimal readout frequency.